docs/vector/vector-intro.md

```diff
@@ -45,28 +46,28 @@ val nonIntersection = myPoint & Point(200, 300)
 
 At this point, the values `selfIntersection` and `nonIntersection`
 are `GeometryResult` containers. These containers are what many JTS
-operations on `Geometry` objects will wrap their results in. Source
-for the To idiomatically destructure these wrappers, we do the following:
+operations on `Geometry` objects will wrap their results in.
+To idiomatically destructure these wrappers, we can use the
+`as[G <: Geometry]` function which either returns `Some(G)` or `None`.
```

Comment:
`as` is convenient but it gives up some typesafety - `(r: PointGeometryIntersectionResult).as[Line]` will compile fine but has zero possibility of returning a `Some` .  I think we should at least mention the benefit of the `match` based approach (exhaustiveness checking) as this is the entire reason the Result types exist.